src/patternfly/components/NotificationBadge/notification-badge.scss

```diff
@@ -1,35 +1,63 @@
 .pf-c-notification-badge {
-  --pf-c-notification-badge--after--BorderColor: var(--pf-global--BackgroundColor--100);
-  --pf-c-notification-badge--after--BorderRadius: var(--pf-global--BorderRadius--lg);
+  --pf-c-notification-badge--Color: var(--pf-global--ZIndex--sm);
+  --pf-c-notification-badge--child--ZIndex: var(--pf-global--ZIndex--sm);
+
+  // Notification badge
+  --pf-c-button--c-notification-badge--MarginRight: calc(var(--pf-global--spacer--xs) * -1);
+  --pf-c-button--c-notification-badge--MarginLeft: calc(var(--pf-global--spacer--xs) * -1);
+
+  // ::after border treatment
+  --pf-c-notification-badge--after--BorderColor: var(--pf-global--BackgroundColor--dark-200);
+  --pf-c-notification-badge--after--BorderRadius: var(--pf-global--BorderRadius--md);
   --pf-c-notification-badge--after--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-notification-badge--after--ZIndex: var(--pf-global--ZIndex--xs);
   --pf-c-notification-badge--after--Top: 0;
   --pf-c-notification-badge--after--Right: 0;
-  --pf-c-notification-badge--after--Width: calc(var(--pf-global--spacer--sm) + var(--pf-c-notification-badge--after--BorderWidth) + var(--pf-c-notification-badge--after--BorderWidth));
-  --pf-c-notification-badge--after--Height: calc(var(--pf-global--spacer--sm) + var(--pf-c-notification-badge--after--BorderWidth) + var(--pf-c-notification-badge--after--BorderWidth));
+  --pf-c-notification-badge--after--Bottom: 0;
+  --pf-c-notification-badge--after--Left: 0;
+  --pf-c-notification-badge--after--Width: auto;
+  --pf-c-notification-badge--after--Height: auto;
   --pf-c-notification-badge--after--BackgroundColor: transparent;
 
   // This translation works for the Bell icon, although adjusted for a happy medium between FF and Chrome.
   // If other icons are used, a modifier may need to be created to allow for proper positioning on that icon.
-  --pf-c-notification-badge--after--TranslateX: 50%;
-  --pf-c-notification-badge--after--TranslateY: calc(var(--pf-c-notification-badge--after--BorderWidth) * -1);
+  --pf-c-notification-badge--after--TranslateX: 0;
+  --pf-c-notification-badge--after--TranslateY: 0;
 
   // The icon is normalized to be square in order to facilitate consistency in positioning
   --pf-c-notification-badge__i--Width: var(--pf-global--spacer--md);
   --pf-c-notification-badge__i--Height: var(--pf-global--spacer--md);
```

Comment:
I think these could be set to `auto` now since we aren't positioning the unread dot in this component. Also with a comment to remove them in a breaking change.